src/shared/utils/ReactNodeTypes.js

```diff
@@ -20,7 +21,7 @@ var ReactNodeTypes = {
   COMPOSITE: 1,
   EMPTY: 2,
 
-  getType: function(node) {
+  getType: function(node: ReactElement): number {
```

Comment:
Instead of returning `number`, could you do

```js
type ReactNodeType =  0 | 1 | 2;
```

```js
getType: function(node: ReactElement<any>): ReactNodeType {
```

this way it'll prevent people from passing arbitrary numbers, flow will force to use the literal 0, 1, 2 or anything that returns those values. It's not perfect but much better.